File: 3d_printer.java
import java.util.*;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int X = sc.nextInt();
		int Y = sc.nextInt();
		int Z = sc.nextInt();
		String[][] S = new String[Z][];
		for (int z = 0; z < Z; z++) {
			S[z] = new String[X];
			for (int x = 0; x < X; x++) {
				S[z][x] = sc.next();
			}
			sc.next();
		}
		while (Z-- > 0) {
			String t = "";
			for (int y = 0; y < Y; y++) {
				char c = '.';
				for (int x = 0; x < X; x++) {
					if (S[Z][x].charAt(y) == '#') {
						c = '#';
						break;
					}
				}
				t += c;
			}
			System.out.println(t);
		}
	}
}

File: 3d_printer.kt
import java.util.*

fun main() {
    val sc = Scanner(System.`in`)
    val X = sc.nextInt()
    val Y = sc.nextInt()
    val Z = sc.nextInt()
    val S = Array(Z) { Array(X) { "" } }

    for (z in 0 until Z) {
        for (x in 0 until X) {
            S[z][x] = sc.next()
        }
        sc.next() // Skip the bar line
    }

    for (z in Z - 1 downTo 0) {
        var t = ""
        for (y in 0 until Y) {
            var c = '.'
            for (x in 0 until X) {
                if (S[z][x][y] == '#') {
                    c = '#'
                    break
                }
            }
            t += c
        }
        println(t)
    }
}
